src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/DiagnosticReportCodeTest.java

```diff
@@ -0,0 +1,60 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+
+class DiagnosticReportCodeTest {
+
+    @Test
+    void testMappingToEdifact() {
+        var expectedValue = "GIS+N'";
+
+        var diagnosticReportCode = DiagnosticReportCode.builder()
+            .code("N")
+            .build();
+
+        assertEquals(expectedValue, diagnosticReportCode.toEdifact());
+    }
+
+    @Test
+    void testThatMappingToEdifactWithEmptyTypeThrowsEdifactValidationException() {
+        var diagnosticReportCode = DiagnosticReportCode.builder()
+            .code("")
+            .build();
+
+        final EdifactValidationException exception = assertThrows(EdifactValidationException.class,
+            () -> diagnosticReportCode.toEdifact());
```

Comment:
Can be replaced with method reference:
```suggestion
            diagnosticReportCode::toEdifact);
```